corehq/apps/domain/views.py

```diff
@@ -1560,10 +1616,18 @@ def post(self, request, *args, **kwargs):
                     ) % software_plan_name
                 )
             else:
-                messages.success(
-                    request, _(
+                if self.is_downgrade_before_minimum:
+                    start_date = self.current_subscription.date_start + datetime.timedelta(days=31)
```

Comment:
If you get the next subscription object and display its ```date_start```, you wouldn't have to do any recalculations.  It would make the code DRYer and more robust to changes.